addons/mail/models/mail_activity.py

```diff
@@ -20,19 +20,19 @@ class MailActivityType(models.Model):
     """ Activity Types are used to categorize activities. Each type is a different
     kind of activity e.g. call, mail, meeting. An activity can be generic i.e.
     available for all models using activities; or specific to a model in which
-    case res_model_id field should be used. """
+    case res_model field should be used. """
     _name = 'mail.activity.type'
     _description = 'Activity Type'
     _rec_name = 'name'
     _order = 'sequence, id'
 
-    @api.model
-    def default_get(self, fields):
-        if not self.env.context.get('default_res_model_id') and self.env.context.get('default_res_model'):
-            self = self.with_context(
-                default_res_model_id=self.env['ir.model']._get(self.env.context.get('default_res_model'))
-            )
-        return super(MailActivityType, self).default_get(fields)
+    def _get_model_selection(self):
+        return [
```

Comment:
I think this will be troubled if you register a new model in a third party module applicable to this domain, isn't it?

Reply:
Why? How is it different than installing any module, it should refresh the list.